Cargo.toml

```diff
@@ -47,6 +47,8 @@ serde = { version = "1.0", optional = true }
 defmac = "0.2"
 quickcheck = { version = "0.7.2", default-features = false }
 rawpointer = "0.1"
+approx = "0.3"
```

Comment:
I'd prefer to avoid a dependency on `approx` because it's used in only one place (`test_mean_with_array_of_floats`); I'd rather change the test:

```rust
assert!((a.mean().unwrap() - expected_mean).abs() <= f64::EPSILON);
```

Reply:
I think `approx` is quite a handy library for writing tests on floating point operations - I definitely foresee it being useful in many other spots in the test suite.

Reply:
Okay, it's fine with me then. `approx` does look useful. I'll submit a PR implementing the `approx` traits for `ArrayBase`.